marketpulse/main/views.py

```diff
@@ -42,5 +45,33 @@ def edit_fxosprice(request, username='', id=None):
                    'mapbox_token': settings.MAPBOX_TOKEN})
 
 
-def fxosprice_home(request):
-    return render(request, 'home.html')
+@login_required
+def view_fxosprice(request):
+    user = request.user
+    contributions = Contribution.objects.filter(user=user.pk)
+    return render(request, 'fxosprice_view.html',
+                  {'user': user, 'contributions': contributions})
+
+
+@login_required
+def delete_fxosprice(request, contribution_pk):
+    user = request.user
+
+    if not Contribution.objects.filter(user=user.pk, pk=contribution_pk).exists():
+        raise Http404()
+
+    Contribution.objects.get(user=user.pk, pk=contribution_pk).delete()
+    messages.success(request, 'Contribution successfully deleted')
+    return redirect(reverse('main:activities'))
+
+
+@login_required
+def activities(request):
+    user = request.user
+    total_contributions = Contribution.objects.filter(user=user.pk).count()
+    contributions = Contribution.objects.all().count()
+    percent = int((float(total_contributions) / float(contributions)) * 100)
```

Comment:
When there is no contribution ``ZeroDivisionError`` is raised. Please try/catch this exception and return ``0`` on exception.